prow/cmd/sinker/main.go

```diff
@@ -138,13 +138,38 @@ func main() {
 }
 
 type controller struct {
-	logger        *logrus.Entry
-	prowJobClient prowv1.ProwJobInterface
-	podClients    []corev1.PodInterface
-	config        config.Getter
+	logger         *logrus.Entry
+	prowJobClient  prowv1.ProwJobInterface
+	podClients     []corev1.PodInterface
+	config         config.Getter
+	currentMetrics *sinkerReconciliationMetrics
```

Comment:
The data in this structure is only useful for one reconciliation loop, right? So I think we can get away with not storing it in the controller and using it only as necessary in the `clean()` method

Reply:
It is also used on the UNIT test which is the only reason that it is stored here, to make it testable.

Reply:
Good point. I think especially as we move to calling Prometheus libraries instead of just logging, it will become harder and harder to test this logic. It's simple, not critical to the process, and can be verified by inspection, so in this case I think we are OK if we do not test it in that way. @cjwagner do you have thoughts on unit-testing that our components emit correct prometheus output?

Reply:
Removed.